launch.md

```diff
@@ -0,0 +1,151 @@
+  * **exceptions**: Optional flag to determine whether exception messages should be logged to the Debug Console. Defaults to true.
+  * **moduleLoad**: Optional flag to determine whether module load events should be logged to the Debug Console. Defaults to true.
+  * **programOutput**: Optional flag to determine whether program output should be logged to the Debug Console. Defaults to true.
+  * **engineLogging**: Optional flag to determine whether diagnostic engine logs should be logged to the Debug Console. Defaults to false.
+  * **trace**: Optional flag to determine whether diagnostic adapter command tracing should be logged to the Debug Console. Defaults to false.
+  * **traceResponse**: Optional flag to determine whether diagnostic adapter command and response tracing should be logged to the Debug Console. Defaults to false.
+
+**visualizerFile**: .natvis file to be used when debugging. See [Natvis syntax reference](http://aka.ms/natvis#Anchor_8) for information on how to create natvis files.
+
+**Example:**
+
+```json
+{
+   "name": "C++ Launch (Windows)",
+   "type": "cppvsdbg",
+   "request": "launch",
+   "program": "C:\\app1\\Debug\\app1.exe",
+   "symbolSearchPath": "C:\\Symbols;C:\\SymbolDir2",
+   "externalConsole": true,
+   "logging": {
+       "moduleLoad": false,
+       "trace": true
+    },
+   "visualizerFile": "${workspaceRoot}/my.natvis"
+}
+```
+
+
+# Configure the target application
+
+The following options enable you to modify the state of the target application when it is launched
+
+**args**: JSON array of command line arguments to pass to the program when it is launched. Example `["arg1", "arg2]`.
+
+**cwd**: Sets the the working directory of the application launched by the debugger.
+
+**environment**: Environment variables to add to the environment for the program. Example: `[ { "name": "squid", "value": "clam" } ]`.
+
+**Example:**
+
+```json
+{
+   "name": "C++ Launch",
+   "type": "cppdbg",
+   "request": "launch",
+   "program": "${workspaceRoot}/a.out",
+   "args": ["arg1", "arg2"],
+   "environment": [{"name": "squid", "value": "clam"}],
+   "cwd": "${workspaceRoot}"
+}
+```
+
+## Customizing GDB or LLDB 
+
+You can change the behavior of GDB or LLDB by setting the following options.
+
+**MIMode**: Indicates the debugger that VS Code will connect to. Must be set to `gdb` or `lldb`.  This is pre-configured on a per-operating system basis and can be changed as needed.
+
+**miDebuggerPath**: The path to the debugger (such as gdb). When unspecified, it will search the operating system's PATH variable for a debugger (GDB on Linux and Windows, LLDB on OS X).
+
+**stopAtEntry**: If set to true, the debugger should stop at the entrypoint of the target (ignored on attach). Default value is `false`.
+
+**setupCommands**: JSON array of commands to execute in order to setup the GDB or LLDB. Example: `"customLaunchSetupCommands": [ { "text": "target-run", "description": "run target", "ignoreFailures": false }]`.
```

Comment:
Experimentation shows me that the element is NOT named "customLaunchSetupCommands", as shown here, but just "setupCommands".  This example (and again down a few lines) is wrong.